src/index.js

```diff
@@ -96,6 +96,8 @@ Onfido.init = (opts) => {
     },
 
     tearDown() {
+      const state = store.getState()
```

Comment:
not going through a selector feels like bad practice. you should get someone else's perspective on it, because I might not be correct.
However, there are benefits with organizing our reads into queries (ask selectors)

Reply:
The root issue here is that redux is being misused a little. The socket should not be stored in the Redux store as it is cannot be serialized - it's a socket instance. The socket instance should not be stored in the store, and instead should be contained elsewhere - most likely a closure created by a thunk (or saga if you were to be using those). This functionality would handle information coming in from the socket and dispatch events accordingly to keep your store updated. The ability to close this socket would be returned as part of the initialization (which would most likely occur when mounting a certain component), stored on the component instance, and called when unmounting.

Doing things this way keeps the store doing what it's supposed to do - storing data :) Storing implementation details of how to store data directly into the store isn't ideal. If nothing else, you're allowing the entire application to mess with that socket!

Reply:
nice feedback!
And your perspective of accessing data from the store without using the selector? (imagine we are talking about serializable data)